src/node/coinstats.cpp

```diff
@@ -6,29 +6,33 @@
 #include <node/coinstats.h>
 
 #include <coins.h>
+#include <crypto/muhash.h>
 #include <hash.h>
 #include <serialize.h>
-#include <validation.h>
 #include <uint256.h>
 #include <util/system.h>
+#include <validation.h>
 
 #include <map>
 
-static void ApplyStats(CCoinsStats &stats, CHashWriter& ss, const uint256& hash, const std::map<uint32_t, Coin>& outputs)
+static void ApplyStats(CCoinsStats &stats, MuHash3072& muhash, const uint256& hash, const std::map<uint32_t, Coin>& outputs)
 {
     assert(!outputs.empty());
-    ss << hash;
-    ss << VARINT(outputs.begin()->second.nHeight * 2 + outputs.begin()->second.fCoinBase ? 1u : 0u);
     for (const auto& output : outputs) {
-        ss << VARINT(output.first + 1);
-        ss << output.second.out.scriptPubKey;
-        ss << VARINT_MODE(output.second.out.nValue, VarIntMode::NONNEGATIVE_SIGNED);
+        COutPoint outpoint = COutPoint(hash, output.first);
+        Coin coin = output.second;
+
+        TruncatedSHA512Writer ss(SER_DISK, 0);
+        ss << outpoint;
+        ss << (uint32_t)(coin.nHeight * 2 + coin.fCoinBase);
```

Comment:
Hm, it's a good question. I have not questioned it beyond the fact that it's a neat way to document height and coinbase in one uint32. fCoinbase doesn't make a difference for BIP30 because those were all coinbases and the duplicated txs are filtered earlier already. So I don't see a reason why we would need the information but if we can have it without taking up any extra space, why not? I think based on the commit message here I would guess the reasoning is that this information should be added to hash and it was added in one to minimize change: https://github.com/bitcoin/bitcoin/commit/d342424301013ec47dc146a4beb49d5c9319d80a